### PR TITLE
Add controller for zip status

### PR DESCRIPTION
**Pull request by rkondratowicz**
plus tests

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/zipfilestatus/ZipFileEvent.java [hunk 1]
```diff
@@ -13,6 +15,7 @@
     public final String container;
 
     @JsonProperty("created_at")
+    @JsonSerialize(using = CustomTimestampSerialiser.class)
     public final Timestamp createdAt;
```
Comment (aliveni) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/zipfilestatus/ZipFileEvent.java]: I've removed `CustomTimestampSerialiser` in the latest pr https://github.com/hmcts/bulk-scan-processor/pull/466
Is it possible to serialize this date field using `@JsonFormat`?

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/zipfilestatus/ZipFileEvent.java]: Looks like this class was not updated in #466 
Whatever gets in later will have to accommodate I guess.

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/controller/ZipStatusControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        mockMvc
+            .perform(get("/zip-files").param("name", "hello.zip"))
+            .andDo(print())
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.envelopes").isEmpty())
+            .andExpect(jsonPath("$.events").isEmpty());
+    }
+
+    private String toIso(Timestamp timestamp) {
+        return DateTimeFormatter.ISO_INSTANT.format(timestamp.toInstant());
```
[the remark this answers is not in the document]
Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/controller/ZipStatusControllerTest.java]: fixed

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/zipfilestatus/ZipFileEvent.java (hunk 1 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/zipfilestatus/ZipFileEvent.java]: (as this also has to be an `Instant` not `Timestamp`

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/controller/ZipStatusControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        );
+
+        List<ZipFileEvent> events = asList(
+            new ZipFileEvent("type0", "container0", Timestamp.from(now().minusSeconds(10))),
+            new ZipFileEvent("type1", "container1", Timestamp.from(now().minusSeconds(15)))
+        );
+
+        given(service.getStatusFor("hello.zip")).willReturn(new ZipFileStatus(envelopes, events));
+
+        mockMvc
```
Comment (nitinprabhu) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/controller/ZipStatusControllerTest.java]: Can we do the assets with actual values what we expect ? I am not sure if the order of envelopes is maintained ?
```
 mockMvc.perform(get("/post/list-all"))
                .andExpect(status().isOk())
                .andExpect(jsonPath("$.*", hasSize(1)))
                .andExpect(jsonPath("$.blogPosts.*", hasSize(2)))
                .andExpect(jsonPath("$.blogPosts[0].id", equalTo(21)))
                .andExpect(jsonPath("$.blogPosts[0].title", equalTo("Spring Boot Testing Improvements")))
                .andExpect(jsonPath("$.blogPosts[0].content", equalTo("New @SpringBootTest and other new annotations those come within Spring Boot 1.4.0.")))
                .andExpect(jsonPath("$.blogPosts[0].author", equalTo("Richard Roe")))
                .andExpect(jsonPath("$.blogPosts[0].createdDate", equalTo(new DateTime(2016, 12, 14, 0, 0).getMillis())))
                .andExpect(jsonPath("$.blogPosts[1].id", equalTo(57)))
                .andExpect(jsonPath("$.blogPosts[1].title", equalTo("Unit Testing")))
                .andExpect(jsonPath("$.blogPosts[1].content", equalTo("Do not push any production code without unit tests.")))
                .andExpect(jsonPath("$.blogPosts[1].author", equalTo("John Doe")))
                .andExpect(jsonPath("$.blogPosts[1].createdDate", equalTo(new DateTime(2016, 11, 15, 0, 0).getMillis())));
```

**Pull request merged**